tools/SetupFlow/DevHome.SetupFlow/ViewModels/Environments/EnvironmentCreationOptionsViewModel.cs

```diff
@@ -0,0 +1,313 @@
+    /// </summary>
+    /// <param name="recipient">The class that should be receiving the request</param>
+    /// <param name="message">The payload of the message request</param>
+    private void OnEnvironmentOptionsViewRequest(EnvironmentCreationOptionsViewModel recipient, CreationOptionsViewPageRequestMessage message)
+    {
+        message.Reply(_renderedAdaptiveCard);
+    }
+
+    /// <summary>
+    /// The review environments view / summary page view will request an adaptive card to display in the UI if it loads after this view model sends out the original RenderedAdaptiveCard message.
+    /// this can happen when the user navigates away from the review page to another page in Dev Home. E.g the settings page, then navigates back to the review page. At this point the review
+    /// page is unloaded when the user navigates away from it. When they navigate back to it, a new view will be created and loaded, so we need to request the adaptive again from this view model.
+    /// </summary>
+    /// <param name="recipient">The class that should be receiving the request</param>
+    /// <param name="message">The payload of the message request</param>
+    private void OnReviewPageViewRequest(EnvironmentCreationOptionsViewModel recipient, CreationOptionsReviewPageDataRequestMessage message)
+    {
+        // Only send the adaptive card if the session has loaded. If the session hasn't loaded yet, we'll send an empty response. The review page should be sent the adaptive card
+        // once the session has loaded in the OnAdaptiveCardUpdated method.
+        if (!IsAdaptiveCardSessionLoaded)
+        {
+            return;
+        }
+
+        message.Reply(_renderedAdaptiveCard);
+    }
+
+    /// <summary>
+    /// When the extension indicates that the session has stopped, we need to get the result json from the session. Once we get this
+    /// we can send a message to the EnvironmentCreationOptionsTaskGroup to let it know that the adaptive card session has ended.
+    /// It will then update its setup tasks with information to create the compute system.
+    /// </summary>
+    /// <param name="sender">The extension session object who stopped the session</param>
+    /// <param name="args">Data payload that contains the users provided input</param>
+    private void OnAdaptiveCardSessionStopped(ExtensionAdaptiveCardSession sender, ExtensionAdaptiveCardSessionStoppedEventArgs args)
+    {
+        // Send message to the EnvironmentCreationOptionsTaskGroup to let it know that the adaptive card session has ended.
+        // the task group will use the ResultJson to create the compute system.
+        WeakReferenceMessenger.Default.Send(new CreationAdaptiveCardSessionEndedMessage(new CreationAdaptiveCardSessionEndedData(args.ResultJson, _curProviderDetails)));
+        sender.Stopped -= OnAdaptiveCardSessionStopped;
```

Comment:
Do we need to unsubscribe from UiUpdate?
```suggestion
        sender.Stopped -= OnAdaptiveCardSessionStopped;
        sender.UiUpdate -= OnAdaptiveCardUpdated;
```